addons/hw_drivers/controllers/driver.py

```diff
@@ -83,3 +87,25 @@ def download_logs(self):
             res = http.send_file(tools.config['logfile'], mimetype="text/plain", as_attachment=True)
             res.headers['Cache-Control'] = 'no-cache'
             return res
+
+    @http.route('/hw_drivers/download_iot_info', type='http', auth='none', methods=['POST'], csrf=False, cors='*')
+    def download_iot_info(self, iot_security_token):
+        if not helpers.check_iot_security_token(iot_security_token):
+            raise Forbidden("Invalid IoT security token")
+
+        iot_info_json_b = str.encode(
+            json.dumps(
+                IoTObjectInfoController.get_all_iot_info(),
+                ensure_ascii=False,
+                default=date_utils.json_default
+            )
+        )
+        res = http.Stream(
+            type='data',
+            data=iot_info_json_b,
+            download_name=f"iot-odoo-info-{gethostname()}-{datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}.json",
+        ).get_response(as_attachment=True)
+        # As it's a CORS request, the Content-Disposition header is not exposed by default, see:
+        # https://stackoverflow.com/a/5837798
```

Comment:
without this lines, the file downloaded name will always fail (which will default to "download.json")